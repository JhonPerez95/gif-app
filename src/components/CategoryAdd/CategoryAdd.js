import React, { useState } from 'react';

const CategoryAdd = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('Bleach');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setCategories((category) => [...category, inputValue]);
  };
  return (
    <form action="" onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

export default CategoryAdd;
