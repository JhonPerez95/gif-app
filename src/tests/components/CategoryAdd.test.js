import React from 'react';
import { shallow } from 'enzyme';
import CategoryAdd from '../../components/CategoryAdd/CategoryAdd';

describe('Pruebas al componente CategoryAdd', () => {
  const setCategory = jest.fn();
  let wrapper = shallow(<CategoryAdd setCategories={setCategory} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoryAdd setCategories={setCategory} />);
  });

  test('Renderizado correctamente del componente <CategoryAdd/>', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'test del input';
    input.simulate('change', { target: { value } });
  });

  test('No debe llamar SetCategories', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategory).not.toHaveBeenCalled();
  });

  test('Debe llamar setCategories y limpiar la caja de text', () => {
    const value = 'test del input';

    // Simular el inputCahnge
    wrapper.find('input').simulate('change', { target: { value } });

    // Simular el submit
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    // SetCategories haya sido llamada && haya sido llamada una vez
    expect(setCategory).toHaveBeenCalled();
    expect(setCategory).toHaveBeenCalledTimes(1);

    // Que se haya llamado con una funcion
    expect(setCategory).toHaveBeenCalledWith(expect.any(Function));

    // El valor del input debe ser 0
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
