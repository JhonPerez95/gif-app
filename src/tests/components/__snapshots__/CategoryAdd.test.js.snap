// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente CategoryAdd Renderizado correctamente del componente <CategoryAdd/> 1`] = `
<form
  action=""
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
